### PR TITLE
allow scanner to tell what style of stream its getting

### PR DESCRIPTION
**Pull request by andymckay**

### CODE REVIEW

**Comment by tofumatt** [comment]
Looks like you've got a test failure in the style: https://travis-ci.org/mozilla/addons-linter/jobs/126754123#L1167

File: src/scanners/base.js [hunk 1]
```diff
@@ -11,7 +26,7 @@ export default class BaseScanner {
     this._defaultRules = [];
     this._parsedContent = null;
     this._rulesProcessed = 0;
-
+
```
Comment (tofumatt) [inline src/scanners/base.js]: This'd be your style nit; trailing whitespace.

**Comment by tofumatt** [comment]
Don't see any tests here; would be good to add one. Even if it's just to verify the method exists.

**Comment by andymckay** [comment]
I fail in style in so many ways.

**Comment by andymckay** [comment]
Fixed style test and added two tests.

**Comment by muffinresearch** [comment]
I think `fileStreamType` being `string` or `stream` is confusing. The old var name was clearer.

**Comment by muffinresearch** [comment]
Fwiw: It doesn't have to be `streamOrString` but it should be something that is closer to representing what a `stream` or `string` arg represents.

**Comment by andymckay** [comment]
`streamOrStringOrChunk` gets awkward.

**Comment by muffinresearch** [comment]
> streamOrStringOrChunk gets awkward.

hah I agree. How about `fileResultType`?

**Comment by andymckay** [comment]
Works for me.

**Comment by andymckay** [comment]
Updated to `fileResultType`.

**Comment by tofumatt** [comment]
r+

**Pull request merged**